src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -142,9 +142,16 @@
         get-br-list (fn [match-str]
                       (map #(Integer/parseInt (second %))
                            (re-seq #"\$(\d+)" match-str)))
+        get-certname-list (fn [rule]
```

Comment:
Having as an outer function for readability might be a good thing since the amount of code in the embedded function is getting a bit long but I'm okay with this as it is.

Since we obviously didn't have any tests that cover out of bounds backreferences for the more complex ACEs that we can do now, it may be worth at least adding a couple of higher-level tests like these ones - https://github.com/puppetlabs/trapperkeeper-authorization/blob/0.6.0/test/puppetlabs/trapperkeeper/services/authorization/authorization_core_test.clj#L215-L250 - which have `certname` in an ACE map entry.